File: backend/src/main/java/todoktodok/backend/member/application/dto/response/MemberDiscussionResponse.java
package todoktodok.backend.member.application.dto.response;

import todoktodok.backend.book.application.dto.response.BookResponse;
import todoktodok.backend.discussion.domain.Discussion;

public record MemberDiscussionResponse(
        Long discussionId,
        BookResponse book,
        String discussionTitle,
        String discussionOpinion
) {

    public MemberDiscussionResponse(final Discussion discussion) {
        this(
                discussion.getId(),
                new BookResponse(discussion.getBook()),
                discussion.getTitle(),
                discussion.getContent()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/discussion/MemberDiscussionResponse.kt
package com.team.todoktodok.data.network.response.discussion

import com.team.domain.model.member.MemberDiscussion
import kotlinx.serialization.Serializable

@Serializable
data class MemberDiscussionResponse(
    val book: BookResponse,
    val discussionId: Long,
    val discussionOpinion: String,
    val discussionTitle: String,
) {
    fun toDomain(): MemberDiscussion =
        MemberDiscussion(
            id = discussionId,
            book = book.toDomain(),
            discussionOpinion = discussionOpinion,
            discussionTitle = discussionTitle,
        )
}
